internal/pkg/agent/application/upgrade/watcher.go

```diff
@@ -257,3 +263,131 @@ func (ch *AgentWatcher) checkFailures() bool {
 	}
 	return false
 }
+
+// Ensure that AgentWatcherHelper implements the WatcherHelper interface
+var _ WatcherHelper = &AgentWatcherHelper{}
+
+type AgentWatcherHelper struct {
+}
+
+func (a AgentWatcherHelper) InvokeWatcher(log *logger.Logger, agentExecutable string, additionalWatchArgs ...string) (*exec.Cmd, error) {
+	return InvokeWatcher(log, agentExecutable, additionalWatchArgs...)
+}
+
+func (a AgentWatcherHelper) SelectWatcherExecutable(topDir string, previous agentInstall, current agentInstall) string {
+	return selectWatcherExecutable(topDir, previous, current)
+}
+
+func (a AgentWatcherHelper) WaitForWatcher(ctx context.Context, log *logger.Logger, markerFilePath string, waitTime time.Duration) error {
+	return waitForWatcher(ctx, log, markerFilePath, waitTime)
+}
+
+func (a AgentWatcherHelper) TakeOverWatcher(ctx context.Context, log *logger.Logger, topDir string) (*filelock.AppLocker, error) {
+	return takeOverWatcher(ctx, log, new(commandWatcherGrappler), topDir, 30*time.Second, 500*time.Millisecond, 100*time.Millisecond)
```

Comment:
do we want this 30*time.Second be configurable? if not let's make it a constant on top of the file so we can easily track it and change it in the future if we have to